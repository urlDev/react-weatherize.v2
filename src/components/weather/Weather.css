.weatherSituation {
    display: flex;
    justify-content: space-between;
    margin: 0 auto;
    font-family: 'Roboto', sans-serif;
}

.cityName {
    font-size: 20px;
    margin-top: 0;
    margin-bottom: 5px;
}

.date {
    margin: 0;
    font-size: 15px;
}

.degree {
    font-size: 50px;
    margin: 0;
    margin-right: 10px;
}

.center {
    margin: auto 0;
}

@media (min-width: 900px) {
    .weatherSituation {
        margin: 0;
        max-width: 60%;
        width: 30%;
    }
    .cityName {
        font-size: 40px;
    }
    .degree {
        font-size: 100px;
    }
    .date {
        font-size: 18px;
    }
}