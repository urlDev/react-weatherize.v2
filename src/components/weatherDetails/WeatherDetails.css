.weather__details {
    color: white;
    font-family: 'Roboto', sans-serif;
    font-weight: 900;
    width: 90%;
    height: 50%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin-left: auto;
    margin-bottom: 20px;
}

.tableDiv {
    height: 50%;
    font-family: 'Roboto', sans-serif;
    font-weight: 100;
}

table {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    width: 100%;
    height: 100%;
    /* margin-left: 10%; */
}

@media (min-width: 900px) {
    .weather__details {
        margin-bottom: 0;
    }
}