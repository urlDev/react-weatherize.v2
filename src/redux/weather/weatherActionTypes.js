const weatherActionTypes = {
  FETCH_WEATHER_BEGIN: 'FETCH_WEATHER_BEGIN',
  FETCH_WEATHER_SUCCESS: 'FETCH_WEATHER_SUCCESS',
  FETCH_WEATHER_ERROR: 'FETCH_WEATHER_ERROR',
  SAVE_LOCAL_STORAGE: 'SAVE_LOCAL_STORAGE',
};

export default weatherActionTypes;
