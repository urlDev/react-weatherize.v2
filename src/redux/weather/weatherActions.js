/* eslint-disable indent */
import { toast } from 'react-toastify';
import axios from 'axios';

export const fetchWeatherBegin = () => ({
  type: 'FETCH_WEATHER_BEGIN',
});

export const fetchWeatherSuccess = (weather) => ({
  type: 'FETCH_WEATHER_SUCCESS',
  payload: { weather },
});

export const fetchWeatherError = (error) => ({
  type: 'FETCH_WEATHER_ERROR',
  payload: { error },
});

export const fetchWeather = (input = 'Helsinki') => async (dispatch) => {
  dispatch(fetchWeatherBegin());
  try {
    const response = await axios.get(
      `https://cors-anywhere.herokuapp.com/http://api.openweathermap.org/data/2.5/weather?q=${input}&APPID=${process.env.WEATHER_API}&units=metric`,
    );
    const data = await response.data;
    return data.cod === 200
      ? dispatch(fetchWeatherSuccess(data))
      : [
          toast.error(
            data.message &&
              data.message.slice(0, 1).toUpperCase() + data.message.slice(1),
          ),
          dispatch(fetchWeatherError(data.message)),
        ];
  } catch (error) {
    return dispatch(fetchWeatherError(error));
  }
};
