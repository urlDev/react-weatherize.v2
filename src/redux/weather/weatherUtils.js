/* eslint-disable implicit-arrow-linebreak */
/* eslint-disable no-confusing-arrow */

const prevSearchedCities = (prevSearchedCity, searchedCity) =>
  prevSearchedCity.includes(searchedCity)
    ? prevSearchedCity
    : [...prevSearchedCity, searchedCity];

export default prevSearchedCities;
