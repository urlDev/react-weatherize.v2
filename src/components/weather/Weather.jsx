import React from 'react';
import moment from 'moment';
import { useSelector } from '../../tests/__mocks__/reactReduxHooks';

import logo from '../../assets/logo.png';
import './Weather.css';

const Weather = () => {
  const weather = useSelector((state) => state.weather.weather);
  const loading = useSelector((state) => state.weather.loading);

  return (
    <div className="weatherSituation">
      {loading ? (
        <img alt="logo" src={logo} className="loading" />
      ) : (
        <>
          <div className="center">
            <h1 className="degree">
              {parseInt(weather.weather.main.temp, 10)}Â°
            </h1>
          </div>
          <div className="center">
            <h1 className="cityName">{weather.weather.name}</h1>
            <h3 className="date">{moment().format('DD.MM.YYYY')}</h3>
          </div>
          <div className="center">
            <img
              src={`https://openweathermap.org/img/wn/${weather.weather.weather[0].icon}@2x.png`}
              alt=""
            />
          </div>
        </>
      )}
    </div>
  );
};

export default Weather;
