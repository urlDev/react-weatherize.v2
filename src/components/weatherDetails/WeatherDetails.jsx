/* eslint-disable operator-linebreak */
/* eslint-disable react/jsx-one-expression-per-line */
import React from 'react';
import { useSelector } from '../../tests/__mocks__/reactReduxHooks';

import logo from '../../assets/logo.png';
import './WeatherDetails.css';

const WeatherDetails = () => {
  const weather = useSelector((state) => state.weather.weather);
  const loading = useSelector((state) => state.weather.loading);

  return (
    <div className="weatherDetails">
      <h1 className="tableHeader">Weather Details</h1>
      {loading ? (
        <img alt="logo" src={logo} className="loading" />
      ) : (
        <>
          <table className="tableDiv">
            <tbody>
              <tr>
                <td>Visibility</td>
                <td>
                  {parseInt(weather.weather.visibility, 10) / 1000}
                  .0 km
                </td>
              </tr>
              <tr>
                <td>Humidity</td>
                <td>{weather.weather.main.humidity}%</td>
              </tr>
              <tr>
                <td>Wind</td>
                <td>{weather.weather.wind.speed} km/h</td>
              </tr>
            </tbody>
          </table>
        </>
      )}
    </div>
  );
};

export default WeatherDetails;
