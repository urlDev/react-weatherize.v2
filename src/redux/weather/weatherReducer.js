import weatherActionTypes from './weatherActionTypes';

import prevSearchedCities from './weatherUtils';

const initialState = {
  loading: false,
  weather: [],
  error: null,
  storage: [],
};

const weatherReducer = (state = initialState, action) => {
  switch (action.type) {
    case weatherActionTypes.FETCH_WEATHER_BEGIN:
      return {
        ...state,
        loading: true,
      };
    case weatherActionTypes.FETCH_WEATHER_SUCCESS:
      return {
        ...state,
        loading: false,
        weather: action.payload.weather,
        storage: prevSearchedCities(state.storage, action.payload.weather.name),
      };
    case weatherActionTypes.FETCH_WEATHER_ERROR:
      return {
        ...state,
        loading: false,
        error: action.payload.error,
      };
    default:
      return state;
  }
};

export default weatherReducer;
