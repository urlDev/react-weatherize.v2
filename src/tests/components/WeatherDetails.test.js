/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';

import WeatherDetails from '../../components/weatherDetails/WeatherDetails';

import * as ReactReduxHooks from '../__mocks__/reactReduxHooks';
import { weather } from '../fixtures/weather';
import { mockStore } from '../__mocks__/store';

describe('Weather details component', () => {
  let wrapper;
  let store;

  beforeEach(() => {
    store = mockStore({
      weather,
    });

    jest
      .spyOn(ReactReduxHooks, 'useSelector')
      .mockImplementation((state) => store.getState());

    wrapper = shallow(<WeatherDetails store={store} />);
  });

  test('Should render the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should get the weather details', () => {
    expect(store.getState().weather).toBeTruthy();
  });
});
