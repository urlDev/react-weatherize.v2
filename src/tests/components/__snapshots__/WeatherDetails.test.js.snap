// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather details component Should render the component correctly 1`] = `
<div
  className="weatherDetails"
>
  <img
    alt="logo"
    className="loading"
    src="logo.png"
  />
</div>
`;
